test.sh

```diff
@@ -1,5 +1,67 @@
 #!/bin/bash
 
 # main test runner for heketi
+# Executes all executable scripts under tests dir
+# in sorted order.
 
-# TODO: add tests here...
+vecho () {
+	if [ "$verbose" = "yes" ] ; then
+		echo "$*"
+	fi
+}
+
+CLI="$(getopt -o xvh --long exitfirst,verbose,help -n $0 -- "$@")"
+eval set -- "$CLI"
+while true ; do
+	case "$1" in
+		-x|--exitfirst)
+			exitfirst=yes
+			shift
+		;;
+		-v|--verbose)
+			verbose=yes
+			shift
+		;;
+		-h|--help)
+			echo "$0 [options]"
+			echo "  Options:"
+			echo "    -x|--exitfirst    Exit on first test failure"
+			echo "    -h|--help         Display help"
+			exit 0
+		;;
+		--)
+			shift
+			break
+		;;
+		*)
+			echo "unknown option" >&2
+			exit 2
+		;;
+	esac
+done
+
+SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
+
+# environment vars exported for test scripts
+# (this way test scripts dont need cli parsing, we do it here)
+export HEKETI_TEST_EXITFIRST=$exitfirst
+export HEKETI_TEST_SCRIPT_DIR="$SCRIPT_DIR"
+
+cd "$SCRIPT_DIR"
+failed=0
+for tname in $(ls tests | sort) ; do
```

Comment:
We should not take all files that are executable. It will for instance run backup files (e.g of vim) of scripts.
In gluster-kubernetes I run everything that matches test_*.sh Here you may want to use another pattern but almost certainly at minimum *.sh